src/cairo/runner/run.rs

```diff
@@ -94,14 +95,28 @@ pub fn run_program(
 
     let data_len = cairo_runner.get_program().data_len();
 
-    Ok((register_states, cairo_mem, data_len))
+    // get range start and end
+    let range = vm
+        .get_range_check_builtin()
+        .map(|builtin| {
+            let (idx, stop_offset) = builtin.get_memory_segment_addresses();
+            let stop_offset = stop_offset.unwrap_or_default();
+            let rangecheck_base =
+                (0..idx).fold(1, |acc, i| acc + vm.get_segment_size(i).unwrap_or_default());
+            let range_check_end = rangecheck_base + stop_offset;
+
+            (rangecheck_base, range_check_end)
+        })
+        .ok();
+
+    Ok((register_states, cairo_mem, data_len, range))
 }
 
 pub fn generate_prover_args(
     file_path: &str,
     rc_builtin_range: Option<Range<u64>>,
 ) -> (TraceTable<Stark252PrimeField>, CairoAIR, PublicInputs) {
-    let (register_states, memory, program_size) =
+    let (register_states, memory, program_size, _rg_init_end) =
```

Comment:
if we are getting the range of memory addresses from the cairo vm, we should not ignore the variable `rg_init_end`. 
`generate_prover_args` should not receive the hardcoded`rc_builtin_range` now. That value should be calculated with the feature added in this PR.

Reply:
Also, same comment as before, let's be consistent with the namings

Reply:
fixed! thanks!